lib/navigation/preprocess_imu.cpp

```diff
@@ -1,56 +1,128 @@
 #include "preprocess_imu.hpp"
 
+#include <cmath>
+
+#include <algorithm>
+#include <numeric>
+#include <optional>
+
 namespace hyped::navigation {
 
 ImuPreprocessor::ImuPreprocessor()
 {
   // TODOLater: implement
 }
 
-core::ImuData ImuPreprocessor::processData(const core::RawImuData raw_imu_data)
+std::optional<core::ImuData> ImuPreprocessor::processData(const core::RawImuData raw_imu_data)
+{
+  core::ImuData imu_data;
+  core::ImuData accelerometer_data;
```

Comment:
This should be defined at the same time as it is used. It is not allocated in a loop so no point defining up here and assigning 10 lines later.

Reply:
Also when assigned it can then be `const`.